### PR TITLE
Refactor tests force report computed task

### PR DESCRIPTION
**Pull request by kbeker**
Refactoring to tests from first use case

### CODE REVIEW

File: concent_api/core/tests/test_integration_report_computed_task.py [hunk 1]
```diff
@@ -807,75 +900,76 @@ def test_requestor_sends_reject_report_computed_task_and_then_sends_ack_report_c
...
                 reverse('core:send'),
                 data                           = serialized_ack_report_computed_task,
                 content_type                   = 'application/octet-stream',
-                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
             )
 
-        self.assertEqual(response_4.status_code,  400)
-        self.assertIn('error', response_4.json().keys())
+        self._test_400_response(response_4)
```
Comment (Jakub89) [inline concent_api/core/tests/test_integration_report_computed_task.py]: This part of the test is broken, it returns {"error": "Message timestamp too far in the future."}, but it suppose to raise:
```
raise Http400(
                "Received AckReportComputedTask but RejectReportComputedTask "
                "or another AckReportComputedTask for this task has already been submitted."
            )
```

File: concent_api/core/tests/utils.py [hunk 2]
```diff
@@ -81,17 +81,20 @@ def _get_deserialized_report_computed_task(  # pylint: disable=no-self-use
...
-            compute_task_def['deadline'] = self._parse_iso_date_to_timestamp(deadline)
-        else:
-            compute_task_def['deadline'] = deadline or self._parse_iso_date_to_timestamp(self._get_timestamp_string())
+        if compute_task_def is None:
+            compute_task_def                = message.ComputeTaskDef()
+            compute_task_def['task_id']     = task_id
+            if isinstance(deadline, str):
+                compute_task_def['deadline'] = self._parse_iso_date_to_timestamp(deadline)
+            else:
+                compute_task_def['deadline'] = deadline or self._parse_iso_date_to_timestamp(self._get_timestamp_string())
```
Comment (rwrzesien) [inline concent_api/core/tests/utils.py]: This conditions should be separated, please rewrite this function to
```
if isinstance(deadline, int):
    compute_task_def['deadline'] = deadline
elif isinstance(deadline, str):
    compute_task_def['deadline'] = self._parse_iso_date_to_timestamp(deadline)
else:
    compute_task_def['deadline'] = self._parse_iso_date_to_timestamp(self._get_timestamp_string())
```

File: concent_api/core/tests/test_integration_report_computed_task.py [hunk 3]
```diff
@@ -924,47 +1027,41 @@ def test_requestor_sends_ack_report_computed_task_after_deadline_passed(self):
...
                 reverse('core:send'),
                 data                           = serialized_ack_report_computed_task,
                 content_type                   = 'application/octet-stream',
-                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
             )
 
-        self.assertEqual(response_3.status_code,  400)
-        self.assertIn('error', response_3.json().keys())
+        self._test_400_response(response_3)
```
Comment (Jakub89) [inline concent_api/core/tests/test_integration_report_computed_task.py]: Same here it returns {"error": "Message timestamp too far in the future."}, should raise:
```
raise Http400("Time to acknowledge this task is already over.")
```

File: concent_api/core/tests/test_integration_report_computed_task.py [hunk 4]
```diff
@@ -1,70 +1,69 @@
...
+        )
 
         with freeze_time("2017-12-01 11:01:00"):
             response_1 = self.client.post(
                 reverse('core:send'),
                 data                                = serialized_force_report_computed_task,
                 content_type                        = 'application/octet-stream',
-                HTTP_CONCENT_CLIENT_PUBLIC_KEY      = b64encode(PROVIDER_PUBLIC_KEY).decode('ascii'),
-                HTTP_CONCENT_OTHER_PARTY_PUBLIC_KEY = b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii'),
+                HTTP_CONCENT_CLIENT_PUBLIC_KEY      = b64encode(self.PROVIDER_PUBLIC_KEY).decode('ascii'),
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_report_computed_task.py]: Use `_get_encoded_provider_public_key()`

File: concent_api/core/tests/test_integration_report_computed_task.py [hunk 5]
```diff
@@ -1,70 +1,69 @@
...
+                HTTP_CONCENT_OTHER_PARTY_PUBLIC_KEY = b64encode(self.REQUESTOR_PUBLIC_KEY).decode('ascii'),
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_report_computed_task.py]: Use `self._get_encoded_requestor_public_key()`. All below too.

File: concent_api/core/tests/test_integration_report_computed_task.py [hunk 6]
```diff
@@ -807,75 +900,76 @@ def test_requestor_sends_reject_report_computed_task_and_then_sends_ack_report_c
...
-        deserialized_task_to_compute    = load(serialized_task_to_compute,  PROVIDER_PRIVATE_KEY,   REQUESTOR_PUBLIC_KEY, check_time = False)
-
-        with freeze_time("2017-12-01 11:00:05"):
-            ack_report_computed_task = message.AckReportComputedTask()
-        ack_report_computed_task.task_to_compute = deserialized_task_to_compute
-
-        serialized_ack_report_computed_task = dump(ack_report_computed_task, REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+        serialized_ack_report_computed_task = self._get_serialized_ack_report_computed_task(
+            timestamp = "2017-12-01 11:00:05",
+            ack_report_computed_task = self._get_deserialized_ack_report_computed_task(
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_report_computed_task.py]: Please fix styling.

File: concent_api/core/tests/test_integration_report_computed_task.py [hunk 7]
```diff
@@ -924,47 +1027,41 @@ def test_requestor_sends_ack_report_computed_task_after_deadline_passed(self):
...
-        deserialized_task_to_compute    = load(serialized_task_to_compute,  PROVIDER_PRIVATE_KEY,   REQUESTOR_PUBLIC_KEY, check_time = False)
-
-        with freeze_time("2017-12-01 11:00:15"):
-            ack_report_computed_task = message.AckReportComputedTask()
-        ack_report_computed_task.task_to_compute = deserialized_task_to_compute
-
-        serialized_ack_report_computed_task = dump(ack_report_computed_task, REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+        serialized_ack_report_computed_task = self._get_serialized_ack_report_computed_task(
+            timestamp = "2017-12-01 11:00:15",
+            ack_report_computed_task = self._get_deserialized_ack_report_computed_task(
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_report_computed_task.py]: Please fix styling.

File: concent_api/core/tests/test_integration_report_computed_task.py [hunk 8]
```diff
@@ -1222,32 +1338,40 @@ def test_requestor_sends_ack_report_computed_task_with_message_cannot_compute_ta
...
+        cannot_compute_task = self._get_deserialized_cannot_compute_task(
+            timestamp       = "2017-12-01 10:30:00",
+            task_to_compute = task_to_compute,
+            reason          = message.tasks.CannotComputeTask.REASON.WrongCTD
+        )
 
-        serialized_ack_report_computed_task = dump(ack_report_computed_task, REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+        serialized_ack_report_computed_task = self._get_serialized_ack_report_computed_task(
+            timestamp = "2017-12-01 11:00:05",
+            ack_report_computed_task = self._get_deserialized_ack_report_computed_task(
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_report_computed_task.py]: Please fix styling.

File: concent_api/core/tests/test_integration_report_computed_task.py [hunk 9]
```diff
@@ -1412,50 +1559,60 @@ def test_provider_forces_computed_task_report_and_tries_to_receive_after_deadlin
...
         # STEP 3: Requestor accepts computed task via Concent
 
-        with freeze_time("2017-12-01 11:00:05"):
-            ack_report_computed_task = message.AckReportComputedTask()
-        ack_report_computed_task.task_to_compute = deserialized_task_to_compute
-        serialized_ack_report_computed_task = dump(ack_report_computed_task, REQUESTOR_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+        serialized_ack_report_computed_task = self._get_serialized_ack_report_computed_task(
+            timestamp = "2017-12-01 11:00:05",
+            ack_report_computed_task = self._get_deserialized_ack_report_computed_task(
+                timestamp = "2017-12-01 11:00:05",
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_report_computed_task.py]: Please fix styling.

File: concent_api/core/tests/test_integration_report_computed_task.py [hunk 10]
```diff
@@ -1507,37 +1672,48 @@ def test_provider_forces_computed_task_report_and_tries_to_receive_twice(self):
...
```
Comment (rwrzesien) [inline concent_api/core/tests/test_integration_report_computed_task.py]: Please fix styling.

**Comment by kbeker** [comment]
All fixes commited

**Pull request merged**